kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/InternalBatchRestClient.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.server.api.v1
+
+import java.util.{Map => JMap}
+import java.util.Base64
+
+import scala.collection.JavaConverters._
+
+import org.apache.http.client.config.RequestConfig
+import org.apache.http.impl.client.HttpClientBuilder
+
+import org.apache.kyuubi.client.RestClient
+import org.apache.kyuubi.client.api.v1.dto.{CloseBatchResponse, OperationLog}
+import org.apache.kyuubi.server.http.authentication.AuthSchemes
+import org.apache.kyuubi.service.authentication.InternalSecurityAccessor
+
+/**
+ * The internal batch rest client for batch request redirection and it is shared for all the
+ * requests redirected to the same kyuubi instance.
+ *
+ * @param kyuubiInstance the kyuubi instance host:port.
+ * @param socketTimeout the socket timeout for http client.
+ * @param connectTimeout the connect timeout for http client.
+ */
+class InternalBatchRestClient(kyuubiInstance: String, socketTimeout: Int, connectTimeout: Int) {
```

Comment:
can we reuse the new added rest-client ?

Reply:
yes, I have reused the RestClient.

Reply:
Let me try to extend KyuubiRestClient.

Reply:
hmm, I mean KyuubiRestClient and BatchRestApi